### PR TITLE
SpellChecker Sebastian Podgórski

### PR DESCRIPTION
**Pull request by wenaro**

### CODE REVIEW

**Review by orchowski — APPROVED** [review]

**Review by orchowski — APPROVED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java [hunk 1]
```diff
@@ -69,7 +66,7 @@ private String getNodeName(INamedNode node)
     }
 
     private final String NAME_OF_DETECTED_PROPERTY = "name";
-    private final String STEREOTYPE_ENDING = "�";
+    private final String STEREOTYPE_ENDING = "�";
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java]: Use unicode for this character (\uXXXX format)

File: violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import org.languagetool.JLanguageTool;
+import org.languagetool.language.BritishEnglish;
+import org.languagetool.rules.Rule;
+import org.languagetool.rules.RuleMatch;
+
+/**
+ * This class check correctness of string
+ *
+ * @author Sebastian Podgorski
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java]: This should be removed

**Review by abizukojc — CHANGES_REQUESTED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ private String getNodeName(INamedNode node)
     }
 
     private final String NAME_OF_DETECTED_PROPERTY = "name";
-    private final String STEREOTYPE_ENDING = "�";
+    private final String STEREOTYPE_ENDING = "»";
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java]: Write this character in the unicode format e.g \uXXXX
https://unicode-table.com/en/

File: violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java (hunk 2 above)
Comment (wenaro) [reply violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java]: I think it isn't the only one class with @author annotation, so please check all classes for this annotation

**Review by abizukojc — APPROVED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java (hunk 1 above)
Comment (wenaro) [reply violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedDuplicateBehavior.java]: This is not my change, please talk to author of this class

File: violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java (hunk 2 above)
Comment (wenaro) [reply violet-framework/src/main/java/com/horstmann/violet/workspace/spellchecker/SpellChecker.java]: Done

**Comment by RoughTomato** [comment]
conflicts need to be resolved before merging

**Review by RoughTomato — CHANGES_REQUESTED** [review]
Merge conflicts need to be resolved before merging can be done.

**Review by abizukojc — CHANGES_REQUESTED** [review]
This PR shouldn't be merged now. There are too much files modified and most next PRs will have conflicts after this merge.

**Review by NdV66 — CHANGES_REQUESTED** [review]
I agree  with @abizukojc

**Review by orchowski — CHANGES_REQUESTED** [review]
Conflicts again. 
Resolve it, or (which will be much better) remove this useless changes, because you're conflicting with everything in this case.

**Comment by wenaro** [comment]
Optimize imports is needed, because it is better to have imported one package e.g java.awt.Color
then all packages from library like java.awt.*. Conflicts are resolved on a constant basis

**Review by orchowski — APPROVED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java [hunk 4]
```diff
@@ -224,6 +229,41 @@ public void propertyChange(final PropertyChangeEvent event)
...
+     * @param nodeName with html tags
+     * @return clear node name without html tags
+     */
+    @NotNull
+    private String[] getClearNodeName(String nodeName) {
+
+        final String regexSplitByUpperCase = "(?=\\p{Lu})";
+
+        nodeName = nodeName.replaceAll("<html><font size=\\+1>", "").replaceAll("</font><html>", "");
+        nodeName = nodeName.replaceAll("<html><center>�interface�</center> <font size=\\+1>", "");
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java]: for chars << and >> use unicodes

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java [hunk 5]
```diff
@@ -224,6 +229,41 @@ public void propertyChange(final PropertyChangeEvent event)
...
+     *
+     * @param nodeName with html tags
+     * @return clear node name without html tags
+     */
+    @NotNull
+    private String[] getClearNodeName(String nodeName) {
+
+        final String regexSplitByUpperCase = "(?=\\p{Lu})";
+
+        nodeName = nodeName.replaceAll("<html><font size=\\+1>", "").replaceAll("</font><html>", "");
```
Comment (orchowski) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java]: coding UTF8

**Review by abizukojc — CHANGES_REQUESTED** [review]
Fix encoding problems

**Review by orchowski — APPROVED** [review]

File: violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java [hunk 6]
```diff
@@ -256,10 +256,10 @@ private void checkCorrectnessOfString(final INamedNode edited) {
     private String[] getClearNodeName(String nodeName) {
 
         final String regexSplitByUpperCase = "(?=\\p{Lu})";
-
+        System.out.println(nodeName);
         nodeName = nodeName.replaceAll("<html><font size=\\+1>", "").replaceAll("</font><html>", "");
-        nodeName = nodeName.replaceAll("<html><center>�interface�</center> <font size=\\+1>", "");
-        nodeName = nodeName.replaceAll("<html><center>�enumeration�</center> <font size=\\+1>", "");
+        nodeName = nodeName.replaceAll("<html><center>«interface»</center> <font size=\\+1>", "");
+        nodeName = nodeName.replaceAll("<html><center>«enumeration»</center> <font size=\\+1>", "");
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/EditSelectedBehavior.java]: You should use Unicode notation (\uXXXX)

**Review by abizukojc — APPROVED** [review]

**Pull request merged**